### PR TITLE
feat(purge): remove locked field on purge

### PR DESCRIPTION
**Pull request by stonebuzz**
remove locked field on purge

### CODE REVIEW

**Review by orthagh — APPROVED** [review]

File: hook.php [hunk 1]
```diff
@@ -869,6 +869,8 @@ function plugin_pre_item_purge_glpiinventory($parm)
     $rule = new RuleMatchedLog();
     $rule->deleteByCriteria(['itemtype' => $itemtype, 'items_id' => $items_id]);
 
+    Lockedfield::releaseLockForAsset($itemtype, $items_id);
```
Comment (cedric-anne) [inline hook.php:872]: Why not using `deleteByCriteria` method ?

Comment (trasher) [reply hook.php:872]: Ping @skita? As far as I can see, this would made change to core useless (maybe better).

Comment (stonebuzz) [reply hook.php]: fixed

File: hook.php [hunk 2]
```diff
@@ -869,6 +869,9 @@ function plugin_pre_item_purge_glpiinventory($parm)
     $rule = new RuleMatchedLog();
     $rule->deleteByCriteria(['itemtype' => $itemtype, 'items_id' => $items_id]);
 
+    $lf = new LockedField();
+    $lf->deleteByCriteria(['itemtype' => $itemtype, 'items_id' => $items_id]);
```
Comment (cedric-anne) [inline hook.php:873]: I do not get why it is done in plugin. If an item is purged, why GLPI is not cleaning locks by itself ?

Comment (stonebuzz) [reply hook.php:873]: To avoid GLPI minor version dependencie

Comment (stonebuzz) [reply hook.php:873]: Ok, so this PR is useless now we can close it=)

**Pull request closed without merging**